src/plugin/any23/src/java/org/apache/nutch/any23/Any23ParseFilter.java

```diff
@@ -0,0 +1,155 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nutch.any23;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.Collections;
+
+import org.apache.any23.Any23;
+import org.apache.any23.writer.BenchmarkTripleHandler;
+import org.apache.any23.writer.NTriplesWriter;
+import org.apache.any23.writer.TripleHandler;
+import org.apache.any23.writer.TripleHandlerException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.nutch.metadata.Metadata;
+import org.apache.nutch.parse.HTMLMetaTags;
+import org.apache.nutch.parse.HtmlParseFilter;
+import org.apache.nutch.parse.Parse;
+import org.apache.nutch.parse.ParseResult;
+import org.apache.nutch.protocol.Content;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.DocumentFragment;
+
+/**
+ * <p>This implementation of {@link org.apache.nutch.parse.HtmlParseFilter}
+ * uses the <a href="http://any23.apache.org">Apache Any23</a> library
+ * for parsing and extracting structured data in RDF format from a
+ * variety of Web documents. The supported formats can be found at <a href="http://any23.apache.org">Apache Any23</a>.
+ * <p>In this implementation triples are written as Notation3 e.g.
+ * <code><http://www.bbc.co.uk/news/scotland/> <http://iptc.org/std/rNews/2011-10-07#datePublished> "2014/03/31 13:53:03"@en-gb .</code>
+ * and triples are identified within output triple streams by the presence of '\n'.
+ * The presence of the '\n' is a characteristic specific to N3 serialization in Any23.
+ * In order to use another/other writers implementing the
+ * <a href="http://any23.apache.org/apidocs/index.html?org/apache/any23/writer/TripleHandler.html">TripleHandler</a>
+ * interface, we will most likely need to identify an alternative data characteristic
+ * which we can use to split triples streams.</p>
+ * <p>
+ *
+ */
+public class Any23ParseFilter implements HtmlParseFilter {
+
+  /** Logging instance */
+  public static final Logger LOG = LoggerFactory.getLogger(Any23ParseFilter.class);
+
+  private Configuration conf = null;
+
+  /** Constant identifier used as a Key for writing and reading
+   * triples to and from the metadata Map field.
+   */
+  private final static String ANY23_TRIPLES = "Any23-Triples";
```

Comment:
This needs to be changed from private to something else if it is to be used in later Indexing Filter tasks.